src/common/function/src/system.rs

```diff
@@ -34,14 +34,35 @@ pub(crate) struct SystemFunction;
 
 impl SystemFunction {
     pub fn register(registry: &FunctionRegistry) {
-        registry.register_scalar(BuildFunction);
-        registry.register_scalar(VersionFunction);
-        registry.register_scalar(DatabaseFunction);
-        registry.register_scalar(ReadPreferenceFunction);
-        registry.register_scalar(PgBackendPidFunction);
-        registry.register_scalar(ConnectionIdFunction);
-        registry.register_scalar(TimezoneFunction);
+        registry.register_scalar(BuildFunction::default());
+        registry.register_scalar(VersionFunction::default());
+        registry.register_scalar(DatabaseFunction::default());
+        registry.register_scalar(ReadPreferenceFunction::default());
+        registry.register_scalar(PgBackendPidFunction::default());
+        registry.register_scalar(ConnectionIdFunction::default());
+        registry.register_scalar(TimezoneFunction::default());
         registry.register(ProcedureStateFunction::factory());
         PGCatalogFunction::register(registry);
     }
 }
+
+macro_rules! define_nullary_udf {
+    ($(#[$attr:meta])* $name: ident) => {
+        $(#[$attr])*
+        #[derive(Clone, Debug, derive_more::Display)]
+        #[display("{}", self.name())]
+        pub(crate) struct $name {
+            signature: datafusion_expr::Signature,
+        }
+
+        impl Default for $name {
+            fn default() -> Self {
+                Self {
+                    signature: datafusion_expr::Signature::nullary(Volatility::Immutable),
+                }
+            }
+        }
+    };
+}
```

Comment:
This type of macro is going to be difficult to read and update. Do you know if it's possible to write it as derive macro like `derive(NullarySignature)`

Reply:
Another idea, can we define this signature as a `&'static` constant, so it doesn't have to be member of all these functions.

Reply:
> This type of macro is going to be difficult to read and update. Do you know if it's possible to write it as derive macro like `derive(NullarySignature)`

It's possible, but will make more complicate codes: two procedure macros have to be used together, one for rewriting(or redefining) the struct to add the "signature" field, the other one for implementing the `Default` on it. I think it'd be better to keep the simple `macro_rules` way.